modules/gateways/Blockonomics/Blockonomics.php

```diff
@@ -79,19 +79,51 @@ private function generateCallbackSecret() {
 	 */
 	public function getApiKey() {
 		return Capsule::table('tblpaymentgateways')
-			->where('gateway', 'blockonomics')
-			->where('setting', 'ApiKey')
-			->value('value');
+				->where('gateway', 'blockonomics')
+				->where('setting', 'ApiKey')
+				->value('value');
 	}
 
 	/*
-	 * Get user configured API key from database
+	 * Get list of crypto currencies supported by Blockonomics
 	 */
-	public function getAltcoins() {
-		return Capsule::table('tblpaymentgateways')
-			->where('gateway', 'blockonomics')
-			->where('setting', 'Altcoins')
-			->value('value');
+	public function getSupportedCurrencies() {
+		$all_currencies = array(
+  			'btc' => array(
+			        'name' => 'Bitcoin',
+			        'uri' => 'bitcoin'
+  			),
+  			'bch' => array(
+			        'name' => 'Bitcoin Cash',
+			        'uri' => 'bitcoincash'
+  			)
+  		);
+		return $all_currencies;
+	}
+
+	/*
+	 * Get list of active crypto currencies
+	 */
+	public function getActiveCurrencies() {
+		$active_currencies = array();
+		$blockonomics_currencies = $this->getSupportedCurrencies();
+		foreach ($blockonomics_currencies as $code => $currency) {
+			if($code == 'btc'){
+				$api = Capsule::table('tblpaymentgateways')
+					->where('gateway', 'blockonomics')
+					->where('setting', 'ApiKey')
+					->value('value');
```

Comment:
Remove this. Not necessary. Keep BTC always as active currency

Reply:
Changed in af03268